### PR TITLE
drpcmux: add tests for server interceptors

### PR DESCRIPTION
**Pull request by shubhamdhama**
This commit adds tests for the server interceptors and RPC handling:

- Tests for unary and streaming server interceptors
- Validation of correct interceptor chaining and execution order
- Error handling and propagation through interceptor chains
- Edge cases with empty interceptor chains

### CODE REVIEW

File: drpcmux/handle_rpc_test.go [hunk 1]
```diff
@@ -0,0 +1,662 @@
...
+		r.Equal(impl, srv)
+
+		// Verify that in1 has been modified by the stream interceptor
+		inputMsg, ok := in1.(*mockMessage)
+		r.True(ok, "expected *mockMessage for in1, got %T", in1)
+
+		if inputMsg.Value == "MODIFIED_original_input" {
+			messageModified = true
+		} else {
+			r.Fail("expected modified input value 'MODIFIED_original_input', got '%s'", inputMsg.Value)
```
[the remark this answers is not in the document]
Comment (shubhamdhama) [reply drpcmux/handle_rpc_test.go:363]: I think we are fine here: https://github.com/stretchr/testify/blob/v1.10.0/require/require.go#L501

**Review by cthumuluru-crdb — APPROVED** [review]
LGTM. Eventually, lets change these tests to use generated protobuf code.
